### ISSUE TITLE
Implement specs on top of shred

### ISSUE THREAD

**Issue by torkleyy** [slide-rs/specs#144]
I'm building a crate, [`shred`](https://github.com/torkleyy/shred), at the moment which will provide functionality for dispatching tasks (systems) which have logical interdependencies. `shred` will also implement [aspects](https://github.com/slide-rs/specs/issues/51); with that knowledge it will allow a lock-free system execution.

It will affect a quite a few modules, mostly `planner` and `world`, so I'd recommend we concentrate PRs on storages and parallel iteration.

**Comment by torkleyy** [comment]
I'm not quite sure about the naming yet:

In shred, there are `Task` and `TaskData`. However, these should be `System` and `SystemData` in specs. Should it also be `Task` for specs ( :-1: because it's confusing for an ECS), should we have a re-export in `specs` ( :-1: because the docs are still written for `Task`, custom derive cannot be re-exported) or wrap the stuff in specs ( :-1:  we have to keep both projects in sync, extra derive in specs necessary, documentation has to be written in specs, too). Renaming it in `shred` would probably also be confusing.

**Comment by kvark** [comment]
@torkleyy the naming problems are typically indicating the lack of clarity in understanding the matter in the first place. Here, the semantics of "task" and "system" are different. Task is intended to be fired once, then done, then forgotten. If this is not the case for `shred`, then it should be renamed. A system is supposed to work semi-permanently, thus I think the `specs` naming here makes sense.

**Comment by torkleyy** [comment]
Very good argument, thank you!

**Comment by kvark** [comment]
@torkleyy just to clarify, I didn't mean to say your understanding is weak (it's not!), it was just general observation about the naming problem (that I encounter a lot) that I decided to share.

**Comment by torkleyy** [comment]
@kvark 

No worries, I did not conceive it that way. It's good you brought it up, because now I know how to solve it! I actually thought about `Task` / `System` when writing `shred`; the first one already sounded a bit odd because its usually used for a one-time thing, but I think I decided for the latter one because I had the feeling that naming it `System` would make it look like it's only usable for `specs`. Now that you explained the semantics, it's clear to me that I can name it `System` because it's actually more appropriate.
